### ISSUE TITLE
Modern Websites

### ISSUE THREAD

**Issue by tobiastromm** [rn10950/RetroZilla#6]
Hi.

Please, can we have support to modern sites like Youtube, Facebook?

Thanks.

**Comment by Atavic** [comment]
Most big sites now check browser's user agent and force you to the latest and lamest firefox, chrome or MS-edge. Workarounds are still possible by changing UAString via local proxy.

**Comment by brunocastello** [comment]
> Most big sites now check browser's user agent and force you to the latest and lamest firefox, chrome or MS-edge. Workarounds are still possible by changing UAString via local proxy.

How?

**Comment by zappy-bit** [comment]
afaik seamonkey can run certain modern websites

**Comment by kristibektashi** [comment]
> Most big sites now check browser's user agent and force you to the latest and lamest firefox, chrome or MS-edge. Workarounds are still possible by changing UAString via local proxy.

That probably won't be enough (but you can always try), Firefox 2 (which RetroZilla is based on) is just way too old. In order for any modern websites to work properly, it's very likely that RetroZilla will need to be updated to a newer Firefox version, which will take time and effort.

**Comment by LeoKids** [comment]
> > Most big sites now check browser's user agent and force you to the latest and lamest firefox, chrome or MS-edge. Workarounds are still possible by changing UAString via local proxy.
> 
> That probably won't be enough (but you can always try), Firefox 2 (which RetroZilla is based on) is just way too old. In order for any modern websites to work properly, it's very likely that RetroZilla will need to be updated to a newer Firefox version, which will take time and effort.

But what about the compability with old OSes this fork is intended for? Firefox 3.0 dropped support for Windows 9x and NT 4.0 which makes it a challenge...

Also,when that comment was made,at the time big sites checked your user agent and would force you to the latest Firefox,Chrome or MS-edge at the time,none of which is compatible with 9x or NT 4.0 but were compatible with XP SP2 or later...Nowadays you just need to upgrade your browser...

**Comment by hron84** [comment]
> But what about the compability with old OSes this fork is intended for? Firefox 3.0 dropped support for Windows 9x and NT 4.0 which makes it a challenge...

There is a difference between your interpretation of compatiiblity and the project's own vision. As far as I understand, RetroZilla does not aim for full compatibility with the modern web, rather it makes you possible to reach websites for e.g. driver or application download, use some web-based GUI interfaces on your local infra, etc. by providing SSL/TLS compatibility, providing some best effort upgrades in JS engine, and so on.

Windows 95/98/ME/2k/XP are unsupported long enough to not be safe for using a generic-purpose internet-browsing operating system. There is a lot of efforts made for achieve this, but still far from being secure or even usable.

With simple wording: do not attempt to use your Windows 98 machine as your primary internet-browsing platform, it would not work and - sadly - probably never will.

**Comment by LeoKids** [comment]
> > But what about the compatibility with old OSes this fork is intended for? Firefox 3.0 dropped support for Windows 9x and NT 4.0 which makes it a challenge...
> 
> There is a difference between your interpretation of compatibility and the project's own vision. As far as I understand, RetroZilla does not aim for full compatibility with the modern web, rather it makes you possible to reach websites for e.g. driver or application download, use some web-based GUI interfaces on your local infra, etc. by providing SSL/TLS compatibility, providing some best effort upgrades in JS engine, and so on.
> 
> Windows 95/98/ME/2k/XP are unsupported long enough to not be safe for using a generic-purpose internet-browsing operating system. There is a lot of efforts made for achieve this, but still far from being secure or even usable.
> 
> With simple wording: do not attempt to use your Windows 98 machine as your primary internet-browsing platform, it would not work and - sadly - probably never will.

"RetroZilla is a fork of Gecko 1.8.1 for improved compatibility on the modern web, with Windows 95 and Windows NT 4.0 in mind. Right now, RetroZilla's rendering capabilities are pretty similar to Firefox 2.0's, but as RetroZilla progresses, so will its capabilities."

It says it aims for improved compatibility with the modern web,with Windows 95 and NT 4.0 in mind.

**Comment by hron84** [comment]
> It says it aims for improved compatibility with the modern web,with Windows 95 and NT 4.0 in mind.

Yeah. But Firefox 2.0 was a bit long time ago. Making Gecko engine compatible with the modern web on these old platforms will require a ton of work and it will be obviously slow since these platforms are "dead" in a lot of terms, the userbase of these platform is painfully small.

**Comment by LeoKids** [comment]
> > It says it aims for improved compatibility with the modern web,with Windows 95 and NT 4.0 in mind.
> 
> Yeah. But Firefox 2.0 was a bit long time ago. Making Gecko engine compatible with the modern web on these old platforms will require a ton of work and it will be obviously slow since these platforms are "dead" in a lot of terms, the userbase of these platform is painfully small.

You may think that however,making it compatible with the modern web should be "easy",all you need to do is make it support TLS 1.2 and 1.3,the biggest roadblock is the SSL handshake,not the HTML5 or anything they can see...Without a supported TLS version and cipher,you can't access modern websites at all. And supporting HTML5 or anything compatible with modern web like ES5/ES6 or CSS3 is a main goal but isn't the biggest roadblock...

**Comment by kristibektashi** [comment]
> > > It says it aims for improved compatibility with the modern web,with Windows 95 and NT 4.0 in mind.
> > 
> > 
> > Yeah. But Firefox 2.0 was a bit long time ago. Making Gecko engine compatible with the modern web on these old platforms will require a ton of work and it will be obviously slow since these platforms are "dead" in a lot of terms, the userbase of these platform is painfully small.
> 
> You may think that however,making it compatible with the modern web should be "easy",all you need to do is make it support TLS 1.2 and 1.3,the biggest roadblock is the SSL handshake,not the HTML5 or anything they can see...Without a supported TLS version and cipher,you can't access modern websites at all. And supporting HTML5 or anything compatible with modern web like ES5/ES6 or CSS3 is a main goal but isn't the biggest roadblock...

RetroZilla already supports TLS 1.3, also most websites nowadays heavily depend on newer JavaScript standards. And without support for those will simply lock you out (see Google).

**Comment by LeoKids** [comment]
> > > > It says it aims for improved compatibility with the modern web,with Windows 95 and NT 4.0 in mind.
> > > 
> > > 
> > > Yeah. But Firefox 2.0 was a bit long time ago. Making Gecko engine compatible with the modern web on these old platforms will require a ton of work and it will be obviously slow since these platforms are "dead" in a lot of terms, the userbase of these platform is painfully small.
> > 
> > 
> > You may think that however,making it compatible with the modern web should be "easy",all you need to do is make it support TLS 1.2 and 1.3,the biggest roadblock is the SSL handshake,not the HTML5 or anything they can see...Without a supported TLS version and cipher,you can't access modern websites at all. And supporting HTML5 or anything compatible with modern web like ES5/ES6 or CSS3 is a main goal but isn't the biggest roadblock...
> 
> RetroZilla already supports TLS 1.3, also most websites nowadays heavily depend on newer JavaScript standards. And without support for those will simply lock you out (see Google).

Yup,What i'm trying to say is that the biggest roadblock to browse the modern web is the SSL handshake,if the handshake fails,either due to lack of ciphers or TLS version,you can't access the modern web at all,pretty much every website today requires TLS 1.2 and later.

While you may be right on the Javascript part,to say most is a exaggeration,not every website will lock you out just cause you don't have newer javascript standards.

**Comment by kristibektashi** [comment]
Yea you're right, most is an exaggeration, however a few major websites most likely do have this issue
